File: eza-v5/backend/core/schemas/standalone.py
# -*- coding: utf-8 -*-
"""
Standalone Mode Schemas
Minimal production-ready response format
"""

from pydantic import BaseModel
from typing import Literal


class StandaloneChatRequest(BaseModel):
    text: str


class StandaloneChatResponse(BaseModel):
    answer: str
    safety: Literal["Safe", "Warning", "Blocked"]
    confidence: float = 0.95

File: frontend/api/standalone.ts
/**
 * Standalone API Client
 */

import { apiRequest } from "./api_client";

export interface StandaloneChatRequest {
  text: string;
}

export interface StandaloneChatResponse {
  answer: string;
  safety: "Safe" | "Warning" | "Blocked";
  confidence: number;
}

export function askStandalone(text: string): Promise<StandaloneChatResponse> {
  return apiRequest<StandaloneChatResponse>(
    "/api/standalone/standalone_chat",
    "POST",
    { text }
  );
}

